library/net.cpp

```diff
@@ -1204,23 +1203,34 @@ static net_state_list_ptr dnet_check_route_table_victims(struct dnet_node *node,
 
 static net_addr_list dnet_reconnect_victims(struct dnet_node *node, int *flags)
 {
-	net_addr_list addrs;
-
 	dnet_pthread_lock_guard locker(node->reconnect_lock);
 
-	addrs.reserve(node->reconnect_num);
+	if (!node->reconnect_num)
+		return net_addr_list();
+
+	size_t addresses_needed = std::min(node->reconnect_parallelism, size_t(node->reconnect_num));
+
+	net_addr_list addrs;
+	addrs.reserve(addresses_needed);
 
+	size_t addresses_collected = 0;
 	struct dnet_addr_storage *ast, *tmp;
 	list_for_each_entry_safe(ast, tmp, &node->reconnect_list, reconnect_entry) {
-		list_del_init(&ast->reconnect_entry);
-		node->reconnect_num--;
+		if (addresses_collected == addresses_needed)
+			break;
 
-		addrs.push_back(ast->addr);
+		// Check if address is already connected, and if not, add it to return list
+		if (!dnet_state_search_by_addr(node, &ast->addr)) {
```

Comment:
`dnet_state_search_by_addr_nolock()` increments refcounter of found state, so it should be decremented here.

Reply:
Thanks, I'll fix.

Reply:
I think, `_search_` isn't good naming. Typically search doesn't mutate any. If owning occur, `_get_` or `_acquire_` must be used. A lot of functions it the project aren't named perfectly :-(

Reply:
Actually it searches `dnet_net_state` by address and increments refcount since it returns a new reference to it